exercises/D-logical-operators/exercise2.js

```diff
@@ -5,7 +5,34 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
+function isNegative( number ) {
+  if (number < 0 ){ 
+    return "number is negative";
+  } else {
+    return "number is not negative";
+  }
+
+}
+
+function range(number) {
+  if ( 5 < number<10) {
+    return range ="isBetween5and10";
+  }
+}
+
+function ShortName(name) {
+  if (typeof (name = number && name<5)){
```

Comment:
`typeof` will return the type of the expression passed in, in this case it will evaluate `name = number && name<5`. This will evaluate `name = number` separately, and `name<5` separately. `name = number` evaulates to `undefined`, `name < 5` evaluates to `false` as name has just been reassigned. Therefore, this will be `typeof undefined` which will return the string `"undefined"`, so this becomes `if("undefined")`, which will always evaluate to `true` as this is a non-zero length string. Therefore, this `if` statement will _always_ be true.